modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/application/ApplicationExtensionsManagerConfigurationBuilder.java

```diff
@@ -53,16 +74,63 @@ protected void doConfigure(MuleContext muleContext) throws Exception {
 
     for (ArtifactPlugin artifactPlugin : artifactPlugins) {
       URL manifestUrl =
-          artifactPlugin.getArtifactClassLoader().findResource("META-INF/" + EXTENSION_MANIFEST_FILE_NAME);
-      if (manifestUrl == null) {
-        continue;
-      }
+          artifactPlugin.getArtifactClassLoader().findResource(META_INF_FOLDER + "/" + EXTENSION_MANIFEST_FILE_NAME);
+      if (manifestUrl != null) {
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Discovered extension " + artifactPlugin.getArtifactName());
+        }
+        ExtensionManifest extensionManifest = extensionManager.parseExtensionManifestXml(manifestUrl);
+        extensionManager.registerExtension(extensionManifest, artifactPlugin.getArtifactClassLoader().getClassLoader());
+      } else {
+        discoverExtensionThroughJsonDescriber(artifactPlugin, extensionManager);
 
-      if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Discovered extension " + artifactPlugin.getArtifactName());
       }
-      ExtensionManifest extensionManifest = extensionManager.parseExtensionManifestXml(manifestUrl);
-      extensionManager.registerExtension(extensionManifest, artifactPlugin.getArtifactClassLoader().getClassLoader());
+    }
+  }
+
+  /**
+   * Looks for an extension using the {@link #MULE_PLUGIN_JSON} file, where if available it will parse it calling the
+   * {@link XmlBasedDescriber} describer.
+   *
+   * TODO(fernandezlautaro): MULE-10876 all this code will be merged within the {@link ExtensionManagerAdapterFactory} where the {@link ExtensionManifest} will be dropped and the correct {@link Describer} will be picked up (either Java or XML based)
+   *
+   * @param artifactPlugin to introspect for the {@link #MULE_PLUGIN_JSON} and further resources from its {@link ClassLoader}
+   * @param extensionManager object to store the generated {@link ExtensionModel} if exists
```

Comment:
done in the last commit, the discovery is done in the ArtifactPluginDescriptorFactory and stores it in the ArtifactPluginDescriptor as an Optional<DescriptorProperty>